crowbar_framework/app/models/service_object.rb

```diff
@@ -1084,6 +1084,17 @@ def apply_role(role, inst, in_queue)
     element_order = new_deployment["element_order"]
 
     #
+    # stop chef daemon on all nodes
+    #
+    nodes_to_freeze_chef_service = new_elements.values.flatten
+    nodes_to_freeze_chef_service.each do |node_name|
+      node = NodeObject.find_node_by_name(node_name)
+      @logger.debug "stopping chef service on #{node_name} before apply_role"
+      node.service :chef, :stop
+      wait_for_chef_clients(node_name)
```

Comment:
It's best to do two loops: one where we stop chef-client everywhere, and then one where we wait. Right now, you might hit this:
* stop on node1, see that it's running on node1 and then wait 5 minutes for it to complete
* stop on node2, and same thing, wait 5 minutes
* etc.

If you do two loops, you won't wait 5+5+5+5+5+... minutes but max(5,5,5,...) minutes.